### PR TITLE
feat(google-docs): Add Google Docs integration for document creation and management

### PR DESCRIPTION
**Pull request by lohitya**
### Description :memo:
- **Purpose**: This PR implements a comprehensive Google Docs integration to enable automated document creation, content insertion, text formatting, and document analysis through the Google Docs API v1. 
- **Approach**: 
    - Implements 6 core actions covering essential document operations (create, read, insert, format, parse)
    - Uses Google Docs API batchUpdate endpoint for all modifications to ensure atomic operations and prevent duplicate content
    - Separates plain text insertion (docs_insert_paragraphs) from markdown insertion (docs_insert_markdown_content) for clear action intent and to prevent agent confusion
    - Excludes tab-related functionality since Google Docs API doesn't support programmatic tab creation
    - Provides flexible docs_batch_update action for advanced formatting (bold, italic, colors, font sizes) with comprehensive examples

**Type of change**

- [x] New feature (non-breaking change which adds functionality)

**Updates**

:point_right:  Added 6 Google Docs actions: docs_create, docs_get, docs_insert_paragraphs, docs_insert_markdown_content, docs_batch_update, docs_parse_structure
:point_right: Implemented OAuth2 authentication with required scope
:point_right: Created comprehensive documentation with usage examples and test suite

### Screenshots :camera:

N/A - Backend integration without UI changes

### Test plan :test_tube:
*Provide guidance for how to QA your proposed changes. This is not only for a test but also useful for a reviewer.*

  Authentication Setup
  - Configure Google Docs integration in Autohive platform
  - Authenticate with Google account via OAuth2
  - Verify required scopes are granted (documents, drive.file)

  Document Creation and Retrieval
  - Instruct agent: "Create a new Google Doc titled 'Test Document'"
  - Verify docs_create action executes successfully and returns documentId
  - Instruct agent: "Show me what's in the Test Document"
  - Verify docs_get retrieves document content with full metadata

  Plain Paragraph Insertion
  - Instruct agent: "Add three paragraphs to Test Document: 'First paragraph', 'Second paragraph', 'Third paragraph'"
  - Verify agent uses docs_insert_paragraphs (not docs_insert_markdown_content)
  - Verify response shows paragraphs_inserted: 3
  - Open document in Google Docs UI to confirm paragraphs appear with proper spacing
  - Test append behavior: Add more paragraphs and verify no duplication

  Markdown Content Insertion
  - Instruct agent: "Create a document with sections: Overview, Details, and Conclusion with text under each"
  - Verify agent uses docs_insert_markdown_content with markdown format
  - Verify response shows correct sections_inserted count
  - Open in Google Docs UI to verify:
    - Headings are styled as HEADING_1
    - Document outline (left panel) shows all sections
    - Paragraphs appear as normal text under headings

  Text Formatting with Batch Update
  - Instruct agent: "Make the first paragraph bold and the last paragraph italic in Test Document"
  - Verify agent:
    - Uses docs_parse_structure to identify text positions
    - Uses docs_batch_update with updateTextStyle requests
  - Test various formatting combinations:
    - Bold + italic together
    - Font size changes (12pt, 14pt, 16pt)
    - Text color changes (red, blue, green)
  - Verify formatting appears correctly in Google Docs UI

  Document Structure Parsing
  - Instruct agent: "What headings and sections are in this document?"
  - Verify docs_parse_structure returns accurate structure with:
    - Element types (heading, paragraph, table)
    - Style information (HEADING_1, NORMAL_TEXT, etc.)
    - Text content and position indices (startIndex, endIndex)
  - Test with documents containing mixed content (headings, paragraphs, tables)

  Action Selection and Error Prevention
  - Test that agent chooses correct action:
    - Plain text without headings → docs_insert_paragraphs
    - Markdown with headings → docs_insert_markdown_content
  - Verify agent doesn't duplicate content when using correct action
  - Test error recovery: If agent uses wrong action, verify it detects error and retries with correct action

  Error Handling
  - Test with invalid document_id - verify error message is descriptive
  - Instruct agent to use docs_insert_markdown_content with plain text (no # headings)
  - Verify error: "No sections found in content"
  - Test with expired/invalid OAuth token - verify authentication error handling
  - Test with documents user doesn't have access to

  Edge Cases
  - Test with empty paragraphs array - verify error handling
  - Test with very long content (1000+ paragraphs) - verify performance
  - Test with special characters and emojis in content
  - Test append: true vs append: false parameter behavior

  Additional Unit Tests
  - Run python tests/test_google-docs.py for automated unit tests (requires test access token)

### Author(s) to check :eyeglasses:
- [ ] Project and all contained modules builds successfully
- [ ] Self-/dev-tested
- [ ] Unit/UI/Automation/Integration tests provided where applicable
- [ ] Code is written to standards
- [ ] Appropriate documentation written (code comments, internal docs)

### Reviewer(s) to check :heavy_check_mark:
- [ ] Project and all contained modules builds successfully
- [ ] Change has been dev-/reviewer-tested, where possible
- [ ] Unit/UI/Automation/Integration tests provided where applicable
- [ ] Code is written to standards
- [ ] Appropriate documentation written (code comments, internal docs)

### CODE REVIEW

File: google-docs/google_docs.py [hunk 1]
```diff
@@ -0,0 +1,619 @@
...
+
+
+def get_auth_headers(context: ExecutionContext) -> Dict[str, str]:
+    """Build authorization headers from the execution context."""
+    access_token = context.auth['credentials']['access_token']
+    return {
+        'Authorization': f'Bearer {access_token}',
+        'Content-Type': 'application/json'
+    }
+
```
Comment (ProRedCat) [inline google-docs/google_docs.py]: This shouldn't be required as the integration SDK handles the bearer auth header in this situation
